src/chips/gens_opn2.cpp

```diff
@@ -62,6 +62,38 @@ int GensOPN2::generateAndMix(int16_t *output, size_t frames)
     return (int)frames;
 }
 
+int GensOPN2::generate32(int32_t *output, size_t frames)
+{
+    enum { maxFramesAtOnce = 256 };
+    int16_t temp[2 * maxFramesAtOnce];
+    for(size_t left = frames; left > 0;) {
+        size_t count = (left < maxFramesAtOnce) ? left : maxFramesAtOnce;
+        std::memset(temp, 0, count * sizeof(int16_t) * 2);
+        chip->run((int)count, temp);
+        for(size_t i = 0; i < 2 * count; ++i)
+            output[i] = temp[i];
+        left -= count;
+        output += 2 * count;
+    }
+    return (int)frames;
+}
+
+int GensOPN2::generateAndMix32(int32_t *output, size_t frames)
+{
+    enum { maxFramesAtOnce = 256 };
+    int16_t temp[2 * maxFramesAtOnce];
+    for(size_t left = frames; left > 0;) {
+        size_t count = (left < maxFramesAtOnce) ? left : maxFramesAtOnce;
+        std::memset(temp, 0, count * sizeof(int16_t) * 2);
+        chip->run((int)count, temp);
+        for(size_t i = 0; i < 2 * count; ++i)
+            output[i] += temp[i];
+        left -= count;
+        output += 2 * count;
```

Comment:
Why not to just make next code?
```cpp
int16_t *temp_p = temp;
for(size_t i = 0; i < 2 * count; ++i)
    *(output++) += *(temp_p++);
left -= count;
```

Reply:
I thought looping on pointers was 90's programming, and we were supposed to upgrade to indices because of 21st century compiler progress.
No? :smile:

(reposting because I clicked a wrong button and I'm not sure you would receive. not much important said anyway)

Reply:
Library is intended to support many compilers, and some of them are may be clunky or outdated, or just to compile for using on older or mobile systems.